src/hooks/useComponentVisible.tsx

```diff
@@ -0,0 +1,25 @@
+import { useState, useEffect } from 'react';
+
+export default function useComponentVisible(
+	ref: React.RefObject<HTMLElement>,
+	initialIsVisible = false
+) {
+	const [isComponentVisible, setIsComponentVisible] = useState(
+		initialIsVisible
+	);
+
+	const handleClickOutside = (event: MouseEvent) => {
+		if (ref.current && !ref.current.contains(event.target as Node)) {
+			setIsComponentVisible(false);
+		}
+	};
```

Comment:
Do you mind if I ask what this does here?

Reply:
For sure! So from the name this handles the click outside mouse event. This is used so that if there is a click outside of the dropdown list when it is open, the dropdown list will close (before the dropdown list would still be open even if you clicked outside the dropdown list). The ref is basically a pointer to the container so it can detect whether click was outside the container or not. Hope that makes sense, if you need anything cleared up feel free to ask away!

Reply:
Oh that makes sense! I did something similar in my modal PR but used `stopPropagation()` instead. Thanks a lot :) maybe that'll be an addition to the modals at some point!

Reply:
Yeah I realized that there was very similar behaviour. A noticeable different though is that when Seiya's dropdown isn't open, there is no overlay in the background, so the hook attaches a click event to the whole page.

Since Seiya doesn't have an overlay we want to allow clicks to other buttons on screen (say a button to open a modal), so this is more logic checking where the click occurred (inside or outside).

Instead for your case you attach the click handler to the modal overlay I am assuming, and that will stop propagation and prevent clicks to underlying elements from working, which is what we want when we are bluring a screen